storage/ndb/src/common/mgmcommon/thr_config.cpp

```diff
@@ -438,12 +441,35 @@ THRConfig::compute_automatic_thread_config(
 
     Uint32 rem_threads = cpu_cnt -
       (ldm_threads + 2 + tc_threads + recv_threads + send_threads);
-    ldm_threads += rem_threads;
+    /* Handle integer division issues and prioritise more send threads */
+    Uint32 loop_count = 0;
+    while (1) {
+      if (rem_threads == 0) break;
+      rem_threads--;
+      send_threads++;
+      if (rem_threads == 0) break;
```

Comment:
It looks almost identical to the code above and below.
The main operations here are a series of -- and ++, so it would be helpful to either make the logic more compact or add some comments explaining what’s going on.